src/app/api/quizzes/[quizId]/questions/[questionNumber]/answers/route.ts

```diff
@@ -0,0 +1,57 @@
+import { NextResponse } from "next/server";
+import { db } from "@/lib/db";
+import { quizMessages } from "@/data/responseMessages";
+import { AnswerModel } from "@/model/AnswerModel";
+import { QuestionModel } from "@/model/QuestionModel";
+
+export async function GET(
+  _req: Request,
+  { params }: { params: { quizId: string; questionNumber: string } }
+) {
+  const { quizId, questionNumber } = await params;
+  const quizIdInt = parseInt(quizId, 10);
+  const questionNumberInt = parseInt(questionNumber, 10);
+
+  if (isNaN(quizIdInt) || isNaN(questionNumberInt)) {
+    return NextResponse.json(
+      { error: quizMessages.invalidQuizzId },
+      { status: 400 }
+    );
+  }
+
+  try {
+    const [rows] = await db.query(
+      "SELECT id, question, quiz_id, number FROM Questions WHERE quiz_id = ? AND number = ?",
+      [quizIdInt, questionNumberInt]
+    );
+    const results = Array.isArray(rows) ? (rows as QuestionModel[]) : [];
+
+    if (results.length === 0) {
+      return NextResponse.json(
+        { error: quizMessages.notFound },
+        { status: 404 }
+      );
+    }
+
+    const question = results[0];
+
+    const [answerRows] = await db.query(
+      "SELECT id, answer, is_correct, question_id FROM Answer WHERE question_id = ?",
```

Comment:
you can optimize this code by retrieving the answers with question_number column instead of question_id column, so you won't have to fetch question before to fetch answers from question